packages/js/src/classic-editor/replacement-variables/configurations.js

```diff
@@ -45,7 +45,22 @@ export const parentTitle = {
 export const primaryCategory = {
 	name: "primary_category",
 	getLabel: () => __( "Primary category", "wordpress-seo" ),
-	getReplacement: () => get( window, "wpseoScriptData.analysis.plugins.replaceVars.replace_vars.primary_category", "" ),
+	getReplacement: () => {
+		// Gets the primary category data from `yoast-seo/editor` store.
+		const primaryCategoryData = select( "yoast-seo/editor" ).getReplaceVars()
+			.filter( replaceVar => replaceVar.name === "primary_category" );
```

Comment:
For the reviewers: here the `getPrimaryTaxonomyId()` selector is not used to retrieve the ID from `yoast-seo/editor` store as somehow it returns `undefined` at this point. We suspect it's because at this point, the selector retrieves the ID from the initial state, and the state is not updated even when there are changes to the primary category. Hence, the `getReplaceVars` selector is used here to retrieve the replacement variables, and from there the primary category is retrieved. If you know a better, cleaner solution feel free to leave a comment.

Reply:
This should work on startup, since the primary category ID on the store is `undefined` at that time. However, if the user selects a different primary category, the primary category does get updated in the store, so we can get it from the store instead.